frontend/packages/dev-console/src/components/pipelineruns/detail-page-tabs/PipelineTaskLogs.tsx

```diff
@@ -31,6 +31,29 @@ interface PipelineTaskLogsState {
   targetHeight: number;
 }
 
+const getRenderContainers = (pod: Pod): { containers: Container[]; stillFetching: boolean } => {
+  const containers: Container[] = _.get(pod, ['spec', 'containers'], []);
+  const containerStatus: ContainerStatus[] = _.get(pod, ['status', 'containerStatuses'], []);
+
+  const containerNames = containers.map((c) => c.name);
+  const sortedContainerStatus = [];
+  containerStatus.forEach((cs) => {
+    const containerIndex = containerNames.indexOf(cs.name);
+    sortedContainerStatus[containerIndex] = cs;
+  });
```

Comment:
This is more or less the new code - everything else is just de-duplication of logic.